🌊 Processing view: Document flyout

flash1293 opened the pull request:
Fixes https://github.com/elastic/kibana/issues/218410

<img width="1374" alt="Screenshot 2025-07-02 at 14 41 59" src="https://github.com/user-attachments/assets/dfb8b816-aca6-496e-8dc8-6c50556f0419" />

<img width="1381" alt="Screenshot 2025-07-02 at 14 42 58" src="https://github.com/user-attachments/assets/4c370e39-45ea-4948-aa26-a5340f537c46" />


This PR adds a document flyout to the enrichment view. It shows a basic table and JSON view as well as a diff viewer which is active only if the user is currently applying changes.

## Behavior

* Allow to expand/hide the document flyout from every row of the enrichment view
* By default, show the table and the JSON view
* If there are simulated samples, also show the diff view
* For consistency, the JSON and the diff view both show the flattened view of the document, not the actual source. Once dotted field name handling gets merged in Elasticsearch, this can be changed to not flatten stuff out anymore (it should happen mostly automatically)

## Technical details

This PR is integrating the unified doc viewer component. However, it's maintaining its own registry because the diff and JSON view are not the default ones:
* The diff view is very custom obviously
* The json view is not the default one because the default one is fetching the document again via its id. This would mostly work but not always because not all data sources we support are necessarily pointing to actual documents. Because of this, a new "raw" JSON viewer is added that takes the raw hit as is instead of loading it again. This is slightly inconsistent with what Discover does
* The doc viewer component only gets the hit. However, we need the original doc and the transformed doc at the same time. To minimize changes to the unified doc viewer, this is passed "around" the component via `additionalDocViewerProps`, which holds the current original sample. It's a bit of a hack, but at least it's explicit
* For the diff viewer, I had to integrate with monaco directly, bypassing all the layers of "code editors" we have introduced, to initialize a diff editor instance instead of a regular editor instance. The code is still manageable because it's a pretty simple case (readonly, just one language etc.)

## Questionable choices

* For implementing the diff viewer and json viewer I had to duplicate a little bit of functionality to calculate the height of the monaco editor. I don't like this but I found this to be the least intrusive way to make it work. Happy to change if someone has a better idea, but I don't want to introduce big abstraction layers just for this use case.
* I had to order the keys in the json to get a stable diff because the diffing happens per line, not on the actual json, and the order of keys can change. It's definitely possible to solve this in a better way, but then we wouldn't be able to just use the monaco diff editor, which would add a lot of complexity - I think it's just not worth it to do more
* The doc viewer only updates if the id of the passed in document changes. However, in our case the document can change due to other reasons (change of processors). To convince the doc viewer to update I calculate a new id if a new batch of simulated documents is passed down. To retain the selected record, I keep track of the index as well and always restore from there
* State management is in react hooks - it's starting to get messy but I don't feel like it's worth pulling this into the state machine just yet. Feels "local" enough

## Small changes rationale

* Changed `selectPreviewDocuments` to `selectPreviewRecords` because they aren't documents - documents have an id and an index and stuff. Those are just JSON blobs
* To be able to find the matching original samples for the simulated samples, I refactored a bit the selectors. Not sure whether pretty

flash1293 committed:
wip

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

flash1293 committed:
Merge remote-tracking branch 'upstream/main' into flash1293/document-flyout

flash1293 committed:
revert

flash1293 committed:
clean up

flash1293 committed:
fix

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

flash1293 committed:
fix

flash1293 committed:
Merge branch 'flash1293/document-flyout' of github.com:flash1293/kibana into flash1293/document-flyout
 the commit.

flash1293 committed:
yet another fix

flash1293 commented:
Feedback:
* Removing line numbers
* Use the right colors for red and green

flash1293 committed:
fix tests and themes

flash1293 commented:
@patpscal Adjusted the look of the editor a bit. Are the colors good? I tried to pick the right ones but I don't like how the contrast between the highlighted part in the added/removed line and the regular background is so low...

<img width="596" alt="Screenshot 2025-07-02 at 17 28 06" src="https://github.com/user-attachments/assets/872ef457-6754-4419-9c2f-24247f053076" />

Maybe it's OK though

flash1293 committed:
adjust colors

flash1293 commented:
Changed colors again a bit:

<img width="559" alt="Screenshot 2025-07-02 at 17 58 54" src="https://github.com/user-attachments/assets/0ebbf0b2-b030-4e3c-84fb-b06fab713df8" />

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/315090)
* Commit: 865b4407a45c5a8506af9038faee6a75eccd38a4
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-226149)

### Failed CI Steps
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/315090#0197cbfd-7500-48d1-aa20-471beaf983ed)
* [FTR Configs #84](https://buildkite.com/elastic/kibana-pull-request/builds/315090#0197cc25-cf6a-4ec2-ab78-7f1522bef4e5)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/315090#0197cbfd-7500-48d1-aa20-471beaf983ed) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/315090/jobs/0197cbfd-7500-48d1-aa20-471beaf983ed/artifacts/0197cc25-b7a1-4999-ae0c-09b79a73ac72) FTR Configs #<span></span>84 / serverless observability UI - feature flags ES Query rule - consumers only infrastructure privileges navigates to the rules page
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/315090#0197cc25-cf6a-4ec2-ab78-7f1522bef4e5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/315090/jobs/0197cc25-cf6a-4ec2-ab78-7f1522bef4e5/artifacts/0197cc32-15c2-4f4a-8b4c-0ad652aa2149) FTR Configs #<span></span>84 / serverless observability UI - feature flags ES Query rule - consumers only logs privileges navigates to the rules page

### History
* :broken_heart: [Build #315002](https://buildkite.com/elastic/kibana-pull-request/builds/315002) failed 686f300b41008974fb0ad7fdd9433ced040b26c7
* :broken_heart: [Build #314960](https://buildkite.com/elastic/kibana-pull-request/builds/314960) failed 7a0159514b5b31afea14114162caaa1139852df3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315090","number":315090,"commit":"865b4407a45c5a8506af9038faee6a75eccd38a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315002","number":315002,"commit":"686f300b41008974fb0ad7fdd9433ced040b26c7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314960","number":314960,"commit":"7a0159514b5b31afea14114162caaa1139852df3"}],"number":315090}
buildkite-pr-comment-->

flash1293 committed:
Merge branch 'main' into flash1293/document-flyout

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/315267)
* Commit: 8414add7abfb73982d894b0b704f611a9d546255
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-226149)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/6113a1188a13e3db8034bd2c3ec3447a963a45ce) | [after](https://github.com/elastic/kibana/commit/8414add7abfb73982d894b0b704f611a9d546255) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 471 | 514 | +43 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/6113a1188a13e3db8034bd2c3ec3447a963a45ce) | [after](https://github.com/elastic/kibana/commit/8414add7abfb73982d894b0b704f611a9d546255) | diff |
| --- | --- | --- | --- |
| `streams` | 24 | 29 | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6113a1188a13e3db8034bd2c3ec3447a963a45ce) | [after](https://github.com/elastic/kibana/commit/8414add7abfb73982d894b0b704f611a9d546255) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 556.9KB | 563.0KB | +6.1KB |
| `unifiedDocViewer` | 222.8KB | 222.8KB | +58.0B |
| total |  |  | +6.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/6113a1188a13e3db8034bd2c3ec3447a963a45ce) | [after](https://github.com/elastic/kibana/commit/8414add7abfb73982d894b0b704f611a9d546255) | diff |
| --- | --- | --- | --- |
| `kbnUiSharedDeps-srcJs` | 3.7MB | 3.7MB | +256.0B |
| `streamsApp` | 10.9KB | 11.0KB | +145.0B |
| total |  |  | +401.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/6113a1188a13e3db8034bd2c3ec3447a963a45ce) | [after](https://github.com/elastic/kibana/commit/8414add7abfb73982d894b0b704f611a9d546255) | diff |
| --- | --- | --- | --- |
| `streams` | 25 | 30 | +5 |


</details>


### History
* :broken_heart: [Build #315090](https://buildkite.com/elastic/kibana-pull-request/builds/315090) failed 865b4407a45c5a8506af9038faee6a75eccd38a4
* :broken_heart: [Build #315002](https://buildkite.com/elastic/kibana-pull-request/builds/315002) failed 686f300b41008974fb0ad7fdd9433ced040b26c7
* :broken_heart: [Build #314960](https://buildkite.com/elastic/kibana-pull-request/builds/314960) failed 7a0159514b5b31afea14114162caaa1139852df3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315267","number":315267,"commit":"8414add7abfb73982d894b0b704f611a9d546255"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315090","number":315090,"commit":"865b4407a45c5a8506af9038faee6a75eccd38a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315002","number":315002,"commit":"686f300b41008974fb0ad7fdd9433ced040b26c7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314960","number":314960,"commit":"7a0159514b5b31afea14114162caaa1139852df3"}],"number":315267}
buildkite-pr-comment-->

flash1293 commented:
Discussed with @davismcphee that I'm going to remove the `additionalDocViewerProps` for now and just pass the original sample through via context. This seems like the least intrusive options - none of them are great, but making them great would require a bit more refactoring work on this shared component.

thomheymann reviewed (commented):
UI review only - Will look at the code afterwards.

Already said it during the demo, love the functionality here. I think it's super useful and a really nice improvement! ✨

Couple questions / suggestions that jumped out below. 

---

The font in the code editor is different to the one used in the JSON view. I think the JSON view is the EUI font so would be good to align on that. 

| Diff viewer | JSON |
| --- | --- |
| <img width="431" alt="Screenshot 2025-07-03 at 14 58 31" src="https://github.com/user-attachments/assets/fae35736-1ed5-4c2e-bf3e-b70997156a89" /> | <img width="434" alt="Screenshot 2025-07-03 at 14 58 13" src="https://github.com/user-attachments/assets/626f3a10-291b-4867-b425-057645128f74" /> |

---

It's quite difficult to see which row is expanded. Could we highlight the current row similar to how Discover does it? 

| Streams | Discover |
| --- | --- |
| <img width="981" alt="Screenshot 2025-07-03 at 15 09 05" src="https://github.com/user-attachments/assets/42e46785-3455-4bb3-a937-1f2314577939" /> | <img width="1060" alt="Screenshot 2025-07-03 at 15 03 50" src="https://github.com/user-attachments/assets/f39a0246-cbab-40a2-bf75-54b6077de8fd" /> |

---

General question. Do we need 3 tabs here (Table, Diff, JSON) or would it make more sense to stick to the table/json tabs and then allow users to toggle on/off a diff view within that? I think Diff is just a different way of viewing JSON.

flash1293 committed:
Merge remote-tracking branch 'upstream/main' into flash1293/document-flyout

flash1293 committed:
revert change in doc viewer, fix styling

flash1293 committed:
Merge branch 'flash1293/document-flyout' of github.com:flash1293/kibana into flash1293/document-flyout

flash1293 committed:
selected column background color

flash1293 commented:
Used a context instead of an "additionalProps" prop, data discovery team, you are off the hook here.

@thomheymann Good catch about the font, fixed that and added a row-highlight:

<img width="540" alt="Screenshot 2025-07-04 at 16 31 20" src="https://github.com/user-attachments/assets/00b2f4f9-60f6-4800-a129-bc775b77602c" />
<img width="538" alt="Screenshot 2025-07-04 at 16 32 02" src="https://github.com/user-attachments/assets/a86e82fe-092c-44ee-ba18-6d027c52a152" />

> General question. Do we need 3 tabs here (Table, Diff, JSON) or would it make more sense to stick to the table/json tabs and then allow users to toggle on/off a diff view within that? I think Diff is just a different way of viewing JSON.

@patpscal and @LucaWintergerst discussed this, that was the latest design by my understanding: https://github.com/elastic/kibana/issues/218410#issuecomment-3012658749

tbh I don't have a strong opinion on this - a tab is kind of like a checkbox, too so both work IMHO.

Ready for another look besides that.

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/315890)
* Commit: 61c42a3467a2e9f91f3909a3324871ce2498236d

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/315890#0197d5e0-5051-4ef9-b762-c765b25034a1)

### History
* :green_heart: [Build #315267](https://buildkite.com/elastic/kibana-pull-request/builds/315267) succeeded 8414add7abfb73982d894b0b704f611a9d546255
* :broken_heart: [Build #315090](https://buildkite.com/elastic/kibana-pull-request/builds/315090) failed 865b4407a45c5a8506af9038faee6a75eccd38a4
* :broken_heart: [Build #315002](https://buildkite.com/elastic/kibana-pull-request/builds/315002) failed 686f300b41008974fb0ad7fdd9433ced040b26c7
* :broken_heart: [Build #314960](https://buildkite.com/elastic/kibana-pull-request/builds/314960) failed 7a0159514b5b31afea14114162caaa1139852df3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315890","number":315890,"commit":"61c42a3467a2e9f91f3909a3324871ce2498236d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315267","number":315267,"commit":"8414add7abfb73982d894b0b704f611a9d546255"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315090","number":315090,"commit":"865b4407a45c5a8506af9038faee6a75eccd38a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315002","number":315002,"commit":"686f300b41008974fb0ad7fdd9433ced040b26c7"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/314960","number":314960,"commit":"7a0159514b5b31afea14114162caaa1139852df3"}],"number":315890}
buildkite-pr-comment-->

thomheymann reviewed (approved):
All working as expected. Couple suggestions in the comments below.

Generally speaking I'm a bit concerned about the performance of the data grid and the functionality added here. It takes about 400ms from clicking on the expand button to the selection state being rendered on my laptop (M1). This is a noticeable lag and makes the UI feel really unresponsive. 

It's an existing issue however so I do not want to block this PR since there's a similar delay when switching between different filters (Parsed/Skipped/Failed etc).

flash1293 committed:
Merge remote-tracking branch 'upstream/main' into flash1293/document-flyout

flash1293 committed:
fixed some errors

thomheymann commented on x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/preview_flyout.tsx:
```diff
@@ -0,0 +1,107 @@
+  }, [unifiedDocViewer.registry]);
+
+  const { value: streamDataView } = useAsync(() =>
+    data.dataViews.create({
+      title: streamName,
+      timeFieldName: '@timestamp',
+    })
```
Perfect, yes, agreed. It wasn't obvious to me that this is an optional call.

gsoldevila reviewed (approved):
LGTM (plugin manifest reviewed, one dependency added)

flash1293 commented:
@thomheymann Did you test with the document flyout open or closed? With closed flyout I don't notice any issues. With open flyout it is quite a bit slower, but I don't think it's easily fixable and in-scope for this PR, since it basically depends on what the document flyout is doing internally.

It's really hard to see in your screenshot, but it seems like in the before measurement you switch from a filter with some hits to another one with some hits, while in the after you switch from a filter without any hits to one with hits. In the former case, it can reuse the data grid component instance, while in the later it needs to instantiate from scratch. I think that's where the time difference is coming from.

thomheymann commented:
> @thomheymann Did you test with the document flyout open or closed? With closed flyout I don't notice any issues. With open flyout it is quite a bit slower, but I don't think it's easily fixable and in-scope for this PR, since it basically depends on what the document flyout is doing internally.
> 
> It's really hard to see in your screenshot, but it seems like in the before measurement you switch from a filter with some hits to another one with some hits, while in the after you switch from a filter without any hits to one with hits. In the former case, it can reuse the data grid component instance, while in the later it needs to instantiate it from scratch. I think that's where the time difference is coming from, not from the changes on the PR.

Both tests were carried out in the same way: The flyout was closed and I switched from Parsed (100%) to Partially parsed and back. The partially parsed contains no documents. You can see those in the smaller spikes in the timeline. As soon as the table is rendered that gets bigger (which is expected) but in this PR the render time is doubled and that creates a noticable lag. 

<img width="1608" alt="Screenshot 2025-07-07 at 10 13 57" src="https://github.com/user-attachments/assets/2c68d27d-1b86-4638-8f10-97fa6b4f59bd" />

flash1293 committed:
memoize better

flash1293 commented:
I can't reproduce this - this is current main:

<img width="492" alt="Screenshot 2025-07-07 at 13 02 43" src="https://github.com/user-attachments/assets/d232b105-e3b3-4546-9e17-aa561331c094" />


This is my PR:

<img width="578" alt="Screenshot 2025-07-07 at 13 02 14" src="https://github.com/user-attachments/assets/1b4752e3-6a8f-4b52-820a-058a4af38595" />

The timings fluctuate around 160ms for the whole re-render in both cases.

When trying a couple times, the numbers fluctuate quite a bit, maybe you can get unlucky and data grid decides to re-layout again and sometimes it doesn't or something like that.

I noticed one thing that wasn't memoized properly, but I don't think it makes a difference.

flash1293 commented:
here are the traces - it's 8 vs. 9 skipped frames for me:

[Archive.zip](https://github.com/user-attachments/files/21101397/Archive.zip)

flash1293 committed:
Merge remote-tracking branch 'upstream/main' into flash1293/document-flyout

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/316288)
* Commit: bac655a074dd7a73b64943f2d4638610fb3cfc3e
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-226149)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/7074e96f26e2599501be805aba77e5cf78c95895) | [after](https://github.com/elastic/kibana/commit/bac655a074dd7a73b64943f2d4638610fb3cfc3e) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 497 | 540 | +43 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/7074e96f26e2599501be805aba77e5cf78c95895) | [after](https://github.com/elastic/kibana/commit/bac655a074dd7a73b64943f2d4638610fb3cfc3e) | diff |
| --- | --- | --- | --- |
| `streams` | 25 | 30 | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7074e96f26e2599501be805aba77e5cf78c95895) | [after](https://github.com/elastic/kibana/commit/bac655a074dd7a73b64943f2d4638610fb3cfc3e) | diff |
| --- | --- | --- | --- |
| `streamsApp` | 578.2KB | 584.8KB | +6.6KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/7074e96f26e2599501be805aba77e5cf78c95895) | [after](https://github.com/elastic/kibana/commit/bac655a074dd7a73b64943f2d4638610fb3cfc3e) | diff |
| --- | --- | --- | --- |
| `kbnUiSharedDeps-srcJs` | 3.8MB | 3.8MB | +256.0B |
| `streamsApp` | 10.9KB | 11.0KB | +145.0B |
| total |  |  | +401.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/7074e96f26e2599501be805aba77e5cf78c95895) | [after](https://github.com/elastic/kibana/commit/bac655a074dd7a73b64943f2d4638610fb3cfc3e) | diff |
| --- | --- | --- | --- |
| `streams` | 26 | 31 | +5 |


</details>


### History
* :green_heart: [Build #316125](https://buildkite.com/elastic/kibana-pull-request/builds/316125) succeeded c185f11123f584fa5689c5e4bf0d16e87e9a37fc
* :broken_heart: [Build #315890](https://buildkite.com/elastic/kibana-pull-request/builds/315890) failed 61c42a3467a2e9f91f3909a3324871ce2498236d
* :green_heart: [Build #315267](https://buildkite.com/elastic/kibana-pull-request/builds/315267) succeeded 8414add7abfb73982d894b0b704f611a9d546255
* :broken_heart: [Build #315090](https://buildkite.com/elastic/kibana-pull-request/builds/315090) failed 865b4407a45c5a8506af9038faee6a75eccd38a4
* :broken_heart: [Build #315002](https://buildkite.com/elastic/kibana-pull-request/builds/315002) failed 686f300b41008974fb0ad7fdd9433ced040b26c7
* :broken_heart: [Build #314960](https://buildkite.com/elastic/kibana-pull-request/builds/314960) failed 7a0159514b5b31afea14114162caaa1139852df3

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/316288","number":316288,"commit":"bac655a074dd7a73b64943f2d4638610fb3cfc3e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/316125","number":316125,"commit":"c185f11123f584fa5689c5e4bf0d16e87e9a37fc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315890","number":315890,"commit":"61c42a3467a2e9f91f3909a3324871ce2498236d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315267","number":315267,"commit":"8414add7abfb73982d894b0b704f611a9d546255"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315090","number":315090,"commit":"865b4407a45c5a8506af9038faee6a75eccd38a4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/315002","number":315002,"commit":"686f300b41008974fb0ad7fdd9433ced040b26c7"}],"number":316288}
buildkite-pr-comment-->
